### PR TITLE
ONNX backend test  for constant inputs

### PR DESCRIPTION
**Pull request by tungld**
This patch proposes `check-onnx-backend-constant` to check ONNX backend tests when inputs are constants.
Similar to `check-onnx-backend-dynamic` which turns a static shape into a dynamic shape, `check-onnx-backend-constant` turns model inputs into constant tensors. Constant values are obtained from the real tensors passing to the original model, so they vary on each test run.

`check-onnx-backend-constant` is quite useful to test constant propagation where input tensors are constants.

For example, to change the second input of `add` to constant, we use:
```
TEST_CONSTANT=true IMPORTER_FORCE_CONSTANT="1" TEST_CASE_BY_USER=test_add_cpu make check-onnx-backend
```
and the output model will be:
```mlir
module  {
  func @main_graph(%arg0: tensor<3x4x5xf32>) -> tensor<3x4x5xf32> attributes {input_names = ["x"], output_names = ["sum"]} {
    %0 = "onnx.Constant"() {value = dense<[[[-0.672460437, -0.359553158, -0.813146293, -1.7262826, 0.177426144], [-0.401780933, -1.63019836, 0.462782264, -0.907298386, 0.0519453958], [0.729090571, 0.128982916, 1.13940072, -1.23482585, 0.402341634], [-0.684810101, -0.870797157, -0.578849673, -0.311552525, 0.0561653413]], [[-1.16514981, 0.900826513, 0.46566245, -1.53624368, 1.48825216], [1.89588916, 1.1787796, -0.179924831, -1.07075262, 1.0544517], [-0.403176934, 1.22244501, 0.208274975, 0.976639032, 0.356366396], [0.706573188, 0.0105000203, 1.78587055, 0.126912087, 0.401989371]], [[1.8831507, -1.34775901, -1.27048504, 0.96939671, -1.17312336], [1.94362116, -0.413618982, -0.747454822, 1.92294204, 1.48051476], [1.86755896, 0.906044661, -0.861225664, 1.91006494, -0.268003374], [0.802456378, 0.947251975, -0.155010089, 0.614079356, 0.922206699]]]> : tensor<3x4x5xf32>} : () -> tensor<3x4x5xf32>
    %1 = "onnx.Add"(%arg0, %0) : (tensor<3x4x5xf32>, tensor<3x4x5xf32>) -> tensor<3x4x5xf32>
    return %1 : tensor<3x4x5xf32>
  }
  "onnx.EntryPoint"() {func = @main_graph, numInputs = 1 : i32, numOutputs = 1 : i32, signature = "[    { \22type\22 : \22f32\22 , \22dims\22 : [3 , 4 , 5] , \22name\22 : \22x\22 }\0A\0A]\00@[   { \22type\22 : \22f32\22 , \22dims\22 : [3 , 4 , 5] , \22name\22 : \22sum\22 }\0A\0A]\00"} : () -> ()
}
```

Tests failed when running `check-onnx-backend-constant` (This patch does not fix them):
- and, or, xor: error message "Failed to import ONNX TensorProto due to unsupported data types"
- concat: failed when all inputs are constants for some tests. 
- div: test for 1D tensor failed. test for broadcast passed.
- neg: test for 1D tensor failed, test for 3D tensor passed.

**Syntax-related change**: the current syntax for node tests uses a single variable `test_static_dynamic` to denote static and dynamic tests, which is not scalable in the present of constant tests (and more in the future perhaps), so I changed it to a new dictionary-based syntax.

### CODE REVIEW

File: test/backend/test.py [hunk 1]
```diff
@@ -877,7 +965,12 @@ def run(self, inputs, **kwargs):
             if (sys_is_le != inp_is_le):
                 inputs = list(
                     map(lambda x: x.byteswap().newbyteorder(), inputs))
-            outputs = super().run(inputs)
+            # If constant test, change the model inputs to constants.
+            if args.constant:
+                inputs = self.turn_model_input_to_constant(inputs)
+                self.exec_name = self.compile_model()
```
Comment (chentong319) [inline test/backend/test.py:971]: Can we have `onnx.save(self.model, model_name)` after the model is modified? The saved model may be used for debugging.

Comment (tungld) [reply test/backend/test.py:971]: We did have it. `onnx.save` is inside `compile_model`.

Comment (chentong319) [reply test/backend/test.py:971]: From the name of method and the order of statement, I thought the compile_model is before run_model where the model is changed for constant input.

**Review by chentong319 — APPROVED** [review]
LGTM!

**Comment by AlexandreEichenberger** [comment]
@tungld Please confirm its ready to squash and merge, tx

**Comment by tungld** [comment]
@AlexandreEichenberger Yes, it's ready to merge.

**Pull request merged**